packages/sdk-android/src/deviceManager.ts

```diff
@@ -443,11 +443,48 @@ export const connectToWifiDevice = async (target: string) => {
     }
 
     const deviceResponse = await execCLI(CLI_ANDROID_ADB, connect_str);
+
     if (deviceResponse.includes('connected')) return true;
-    logError(`Failed to ${connect_str}`, { skipAnalytics: true });
+
+    if (deviceResponse.includes('Connection refused')) {
+        logError(`Failed to ${connect_str}. Connection refused. Make sure to that ip and port are correct.`, {
+            skipAnalytics: true,
+        });
+        return false;
+    }
+
+    if (!deviceResponse.includes('Connection refused')) {
+        logWarning(
+            `You'll need to pair your device before installing app. \nFor more information: https://developer.android.com/studio/run/device`
+        );
+        await _pairDevices(target);
+    }
     return false;
 };
 
+const _pairDevices = async (target: string) => {
+    const readline = require('readline').createInterface({
```

Comment:
@Marius456 any reason why are you using `require('readline')` instead of `inquiryPrompt` ?

Reply:
change it to `inquiryPrompt `